src/ioc_scan/ioc_scanner.py

```diff
@@ -158,4 +158,20 @@ def main(blob=None, root="/"):
 
 
 if __name__ == "__main__":
-    sys.exit(main())
+    import argparse
+
+    parser = argparse.ArgumentParser(
+        description="Indicators of compromise (IoC) scanning tool."
+    )
+    parser.add_argument(
+        "-f", "--file", dest="hashfile", help="Get IOC hashes from specified file."
+    )
+    args = parser.parse_args()
+
+    if args.hashes_file:
+        logging.debug("Reading hashes from %s", args.hashfile)
+        with open(args.hashfile) as f:
+            hashblob = f.read()
+        sys.exit(main(hashblob))
+    else:
+        sys.exit(main())
```

Comment:
Why are we putting this here and not in `main()`?

Reply:
Nevermind.  I had to review this more in Atom.   I remember `ioc_scan_cli.py` now.

Reply:
Generally, you put the ArgParser in the section for `if __name__ == "__main__"` so it only gets called if the script is run from the command line and thus has command-line arguments. Then it checks and dispatches validated arguments to the actual program functionality.

Reply:
Looks like someone is having a case of the Mondays... besides me. :) Gonna mark this as resolved.

Reply:
I think I can come up with a counter-argument with regards to code coverage testing when things live outside of main().   

Which reminds me...

Reply:
Typical functionality is `ioc_scanner.main()` being called from `ioc_scan_cli.main()`. So I thought the limited argument parsing should only be used when the script is being run in standalone mode.